org.eclipse.jdt.junit.core/src/org/eclipse/jdt/internal/junit/util/CoreTestSearchEngine.java

```diff
@@ -165,25 +169,36 @@ public static boolean hasJUnit6TestAnnotation(IJavaProject project) {
 	private static boolean hasJUnitJupiterTestAnnotation(IJavaProject project, int junitMajorVersion, IPath... disallowedJunitContainerPaths) {
 		try {
 			if (project != null) {
-				String junitBundlePrefix = JUNIT_PLATFORM_COMMONS_PREFIX;
 				IType type= project.findType(JUnitCorePlugin.JUNIT5_TESTABLE_ANNOTATION_NAME);
 				if (type == null) {
-					junitBundlePrefix = JUNIT_PLATFORM_SUITE_API_PREFIX;
 					type= project.findType(JUnitCorePlugin.JUNIT5_SUITE_ANNOTATION_NAME);
 				}
 				if (type != null) {
 					// check if we have the right JUnit JUpiter version
 					String filename= type.getPath().lastSegment();
-					if ((filename.startsWith(junitBundlePrefix + "_") || filename.startsWith(junitBundlePrefix + "-")) && filename.endsWith(JAR_EXTENSION)) { //$NON-NLS-1$ //$NON-NLS-2$
-						String versionString = filename.substring(junitBundlePrefix.length() + 1, filename.length() - JAR_EXTENSION.length());
-						try {
-							Version version = Version.parseVersion(versionString);
-							if (version.getMajor() != junitMajorVersion) {
-								return false;
+					if (filename.endsWith(JAR_EXTENSION)) {
```

Comment:
Why don't we check first for the "trivial" jar name match now? It would avoid IO for probably most users.